File: Palandrome Test.py
#Nathan Loveless
#Structures of Programming Languges
#12/12/21
string = input("Please enter a string: ") #Prompts for a string
reversed = string[::-1] #Reverses the string
answer = string == reversed #Test if its the same value 
print("The string entered: \n", string, "\nThe string reversed: \n", reversed ) #Displays values
if answer: #If true
    print("The entered string is a palindrome")
else:#If false
    print("The entered string is not a palindrome")
input('Press ENTER to exit') #Closes the program 

File: Structures of Programing Languges Project/Palandrome Test.cpp
//Nathan Loveless
//Structures of Programming Lang
//12/12/21

#include <iostream>
#include <string> //Libraries

using namespace std; //Name space

string getstring() //Function to get string
{
    cout << "Please enter a string: \n"; //Prompts for a string
    string S;  //Creates string s
    cin >> S; //Gets value
    return S; //returns S
}

string palindrome(string S) //Function to test if a plandrome
{
    string P = S; //Copies string
    reverse(P.begin(), P.end()); //Flips
    cout << "The string entered: \n" << S << "\nThe string reversed: \n" << P << "\n";
    if (S == P)  //Tests if they are the same 
    {
        return "The entered string is a palindrome";
    }
    else 
    {
        return "The entered string is not a palindrome";
    }
}

int main() //Main Function 
{
    string S = getstring();//Populates S
    cout << palindrome(S); //Test and displays message
    return 0;
}